### ISSUE TITLE
promptValues: ['login'] and keycloak - You are already authenticated as different user in this session. Please logout first

### ISSUE THREAD

**Issue by morrowyn** [MaikuB/flutter_appauth#197]
Hi, 

When I use 
```
promptValues: ['login'],
```
in combination with keycloak. I can log into just fine. However when I create a different app, using the same keycloak but a different user I get the following the error on the keycloak login screen after a successful login
```
You are already authenticated as different user <user1> in this session. Please logout first
```
Now when I delete the cookies from the chrome web browser app outside my apps. I can log into using user2. But receive the same error when I try to log into the other app with a user1.

Is there a way to delete the stored cookies for keycloak domain/url when using the 
```
promptValues: ['login']
```
since it kind of implies that you want a new session and not re-use the existing sessions through the set cookies.

**Comment by MaikuB** [comment]
The plugin doesn't have a way to do so. This is likely where the ability to call the end session endpoint would help but this needs help from others in the community to implement

**Comment by MaikuB** [comment]
End session support has been added in the 2.0.0-dev.0 prerelease. I'm not familiar with Keycloak but a quick glance would indicate it has a compliant end session endpoint

**Comment by LukasPoque** [comment]
Keycloak has an end session endpoint like stated out here: [Keycloak Doc](https://www.keycloak.org/docs/latest/securing_apps/index.html#logout).

So your request should be something like that:
```
POST http://<auth_server>/auth/realms/<my_realm>/protocol/openid-connect/logout
Headers:
  Content-Type: application/x-www-form-urlencoded
Body:
  client_id=<my_client_id>
  refresh_token=<refresh_token>
```

**Comment by MaikuB** [comment]
2.0.0 has been pushed as a stable release now but that doesn't necessarily mean it'll work with all identity providers. The end session implementation in the native AppAuth Android and IOS SDKs are based on a specific spec that may not be adhered to be everyone. Keycloak seems to have their own implementation so you'd need to look into your own solution for implementing logout

**Comment by MaikuB** [comment]
Hopefully they will. Their docs don't seem to mention the parameters in the spec like `post_logout_redirect_uri`

**Comment by morrowyn** [comment]
The docs mention the following:  
`http://auth-server/auth/realms/{realm-name}/protocol/openid-connect/logout?redirect_uri=encodedRedirectUri` so isn't the encodedRedirectUri the `post_logout_redirect_uri` ?

https://www.keycloak.org/docs/latest/securing_apps/index.html#logout

**Comment by MaikuB** [comment]
No, that is the _value_ of the `redirect_uri` parameter and that parameter that is different to what I wrote. The names of the parameters need to exactly match. This is the point I'm trying to make with regards to [specs](https://openid.net/specs/openid-connect-rpinitiated-1_0.html#RPLogout) being followed. Having an end session ending in the discovery document doesn't mean said endpoint will accept the same parameters. For example, the only parameter Azure AD recognises is `post_logout_redirect_uri` but not the others in the spec I've linked to
